### PR TITLE
Feature/consistency checks

### PR DESCRIPTION
**Pull request by ehuelsmann**
Remaining actions:

* [x] Descriptions of the failed conditions to be added to the check definition files
* [x] Documentation for the Perl module which manages the checks

Closes ledgersmb/LedgerSMB#5212
Closes ledgersmb/LedgerSMB#5206

### CODE REVIEW

File: sql/consistency/00100-cleared-acc_trans-has-recon.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+--- yaml front matter
```
Comment (ylavoie) [inline sql/consistency/00100-cleared-acc_trans-has-recon.sql:1]: frontmatter instead?

File: sql/consistency/00101-transactions-linked-to-arapgl.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+--- yaml front matter
```
Comment (ylavoie) [inline sql/consistency/00101-transactions-linked-to-arapgl.sql:1]: frontmatter?

Comment (ehuelsmann) [reply sql/consistency/00101-transactions-linked-to-arapgl.sql]: Done.

File: sql/consistency/00100-cleared-acc_trans-has-recon.sql (hunk 1 above)
Comment (ehuelsmann) [reply sql/consistency/00100-cleared-acc_trans-has-recon.sql]: Done.

File: sql/consistency/00001-balanced-transactions.sql [hunk 3]
```diff
@@ -5,4 +5,4 @@ title: All transactions are balanced
 select trans_id
   from acc_trans
  group by trans_id
-having sum(amount) <> 0
+having abs(sum(amount_bc)) > 0.005
```
Comment (ylavoie) [inline sql/consistency/00001-balanced-transactions.sql:8]: Should be ok for most currencies

Comment (ehuelsmann) [reply sql/consistency/00001-balanced-transactions.sql:8]:
```sql
having abs(sum(amount_bc)) > POWER(10, -1*(select value::int from defaults where setting_key='decimalplaces'))/2
```
?

Comment (ylavoie) [reply sql/consistency/00001-balanced-transactions.sql:8]: Looks fine to me

**Review by ylavoie — APPROVED** [review]

**Pull request merged**